pkg/prometheus/statefulset.go

```diff
@@ -655,6 +655,9 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *operator.Config, shard in
 	podLabels := map[string]string{
 		"app.kubernetes.io/version": version.String(),
 	}
+	// In cases where a selector label is removed, new sts cannot match existing pods.
```

Comment:
This isn't restricted to removals but even adding or updating a label would trigger a recreation of the statefulset. I think that it should be reinforced that the statefulset's selector is immutable and any change to the list should be carefully evaluated.
Also can we add the same warning to Alertmanager and Thanos?